lib/connextclient/ConnextClient.ts

```diff
@@ -0,0 +1,491 @@
+import assert from 'assert';
+
+import * as connext from '@connext/client';
+import { ConnextStore, FileStorage } from '@connext/store';
+import { IConnextClient } from '@connext/types';
+
+import { SwapClientType, SwapRole, SwapState } from '../constants/enums';
+import { CurrencyInstance } from '../db/types';
+import Logger from '../Logger';
+import swapErrors from '../swaps/errors';
+import SwapClient, {
+  ChannelBalance,
+  ClientStatus,
+  PaymentState,
+  WalletBalance,
+  TradingLimits,
+} from '../swaps/SwapClient';
+import { SwapDeal } from '../swaps/types';
+import { UnitConverter } from '../utils/UnitConverter';
+import { ConnextWallet } from './ConnextWallet';
+import errors, { errorCodes } from './errors';
+import {
+  Channel,
+  ConnextChannelCount,
+  ConnextClientConfig,
+  ConnextInfo,
+  ConnextVersion,
+  TokenPaymentRequest,
+  TokenPaymentResponse,
+} from './types';
+
+const MAX_AMOUNT = Number.MAX_SAFE_INTEGER;
+
+/**
+ * A class representing a client to interact with connext.
+ */
+class ConnextClient extends SwapClient {
+  public readonly type = SwapClientType.Connext;
+  public readonly finalLock = 100;
+  public address?: string;
+  /** A map of currency symbols to token addresses. */
+  public tokenAddresses = new Map<string, string>();
+  private ethProviderUrl: string;
+  private nodeUrl: string;
+  private unitConverter: UnitConverter;
+
+  private client: IConnextClient | undefined;
+  private wallet: ConnextWallet | undefined;
+
+  /**
+   * Creates a connext client.
+   */
+  constructor({
+    config,
+    logger,
+    unitConverter,
+    currencyInstances,
+  }: {
+    unitConverter: UnitConverter;
+    config: ConnextClientConfig;
+    currencyInstances: CurrencyInstance[],
+    logger: Logger;
+  }) {
+    super(logger, config.disable);
+
+    this.ethProviderUrl = config.ethProviderUrl;
+    this.nodeUrl = config.nodeUrl;
+    this.unitConverter = unitConverter;
+    this.setTokenAddresses(currencyInstances);
+  }
+
+  public get minutesPerBlock() {
+    return 0.25; // 15 seconds per block target
+  }
+
+  public get label() {
+    return 'Connext';
+  }
+
+  /**
+   * Checks for connectivity and gets our Connext account address
+   */
+  public initSpecific = async () => {
+    if (!this.wallet) {
+      throw errors.CONNEXT_WALLET_NOT_INITIATED;
+    }
+    if (this.wallet) {
+      await this.initConnextClient();
+      this.setStatus(ClientStatus.Initialized);
+      this.emit('initialized');
+      await this.verifyConnectionWithTimeout();
+    }
+  }
+  /**
+   * Initiates wallet for the Connext client
+   */
+  public initWallet = async (seedMnemonic: string[]) => {
+    this.wallet = new ConnextWallet(seedMnemonic.join(' '));
```

Comment:
Following on the discussion about the mnemonic and wallet initialization, we wouldn't want to create the `ConnextWallet` here. As it's used with lnd, `initWallet` is only called the very first time we create a wallet within lnd. Rather, I'm thinking it makes sense to have the seed as part of the `ConnextClient` constructor.